cincinnati/src/lib.rs

```diff
@@ -187,18 +188,27 @@ impl Graph {
 
     /// Add a transition (edge) from `source` to `target`.
     ///
-    /// Fails with the `WoulcCycle` error if the new edge would lead to a cycle.
+    /// Fails with the `WouldCycle` error if the new edge would lead to a cycle.
     pub fn add_edge(&mut self, from: &ReleaseId, to: &ReleaseId) -> Result<(), Error> {
+        let from_release: String = self.find_by_releaseid(from)?.version().to_string();
+        let to_release: String = self.find_by_releaseid(to)?.version().to_string();
         if self.dag.find_edge(from.0, to.0).is_some() {
             return Err(Error::from(errors::EdgeAlreadyExists {
-                from: self.find_by_releaseid(from)?.version().to_string(),
-                to: self.find_by_releaseid(to)?.version().to_string(),
+                from: from_release,
+                to: to_release,
             }));
         }
 
         self.dag
             .add_edge(from.0, to.0, Empty {})
-            .map(|_| ())
+            .map(|edge| {
+                trace!(
```

Comment:
I'm not a fan of adding log output to the core library. What do you think of returning the edge index from `add_edge` instead, and giving the caller the chance to log it?

Reply:
hmm, right, that would be cleaner

Reply:
Updated, PTAL